docs/runtime/initializing-storage.md

```diff
@@ -0,0 +1,140 @@
+>
+> `_genesis_phantom_data: Default::default(),`
+
+### Enter Values in Configuration File
+
+While there must always be some value specified in code, the deployer may choose to override those values via a configuration file. In Substrate-based chains a chainspec can be generated with `substrate build-spec > chainspec.json`. For our example module, the configuration would look like this.
+
+```json
+{
+  ...
+  "genesis": {
+    "runtime": {
+      ...
+      "my_module": {
+        "ante": 55,
+        "minRaise": 77
+      }
+    }
+  }
+}
+```
+
+## Calculate Individually with `build()`
+
+The third option is ideal when the runtime module developer wants to specify the field's initial value, but it is not known at compile time. For example, it may need to be calculated from other initial values that are deferred to the deployer. To choose this option the runtime developer may include `build(<closure>)` and a [closure](https://doc.rust-lang.org/stable/book/ch13-01-closures.html) that returns the desired initial value.
+
+In our example game, the minimum raise must always be twice the ante. But the ante may be configured by the module deployer.
+```rust
+decl_storage! {
+  trait Store for Module<T: Trait> as TemplateModule {
+    // Points ante'd to play the game
+    pub Ante get(ante) config(): u32;
+
+      // Minimum raise on each turn
+      pub MinRaise get(min_raise) build(|config: &GenesisConfig<T>| config.ante * 2): u32;
+  }
+}
+```
+> A bug in the v1.0 branch, occasionally causes a compilation error when specifying the genesis configuration this way. Adding an additional field will resolve it.
+>
+> `_genesis_phantom_data: Default::default(),`
+
+## Calculate Multiple Values with `add-extra-genesis`
+
+A final option available to runtime developers is to manually add additional fields to the genesis configuration that are not associated with specific storage values. This is useful when all of the initial storage values can be calculated, and they can be calculated from parameters configurable by the deployer. The subtlety here is that the deployer _does_ specify parameters, but those parameters are not, themselves, put into storage. Rather they are used to calculate the values that are put into storage.
+
+In our example game, the module author may want to allow the deployer to specify the smallest number of points worth considering, and calculate both the initial ante and minimum raise from those values.
+
+```rust
+decl_storage! {
+  trait Store for Module<T: Trait> as TemplateModule {
+    // Points ante'd to play the game
+    pub Ante get(ante) build(|config: &GenesisConfig<T>| config.atom * 2): u32;
+
+    // Minimum raise on each turn
+    pub MinRaise get(min_raise) build(|config: &GenesisConfig<T>| config.atom * 3): u32;
+  }
+
+  add_extra_genesis {
+    // The smallest (atomic) number of points worth considering
+    config(atom): u32;
+  }
+}
+```
```

Comment:
maybe it worth noting also that someone can make use of the field build in add_extra_genesis here no ? see srml/staking for example.

Reply:
I've included a link to that part of the staking module and an explanation of what's happening. I think that shows a better use case than doing it in this simple game example which doesn't really need it.